lib/roles/mqtt/mqtt.c

```diff
@@ -1029,17 +1029,27 @@ _lws_mqtt_rx_parser(struct lws *wsi, lws_mqtt_parser_t *par,
 			switch (par->conn_rc) {
 			case 0:
 				goto cmd_completion;
+			/* 3.1.1 errors [MQTT-3.2.3] */
 			case 1:
+				par->reason = LMQCP_REASON_UNSUPPORTED_PROTOCOL;
+				goto send_reason_and_close;
 			case 2:
+				par->reason = LMQCP_REASON_CLIENT_ID_INVALID;
+				goto send_reason_and_close;
 			case 3:
+				par->reason = LMQCP_REASON_SERVER_UNAVAILABLE;
+				goto send_reason_and_close;
 			case 4:
+				par->reason = LMQCP_REASON_BAD_CREDENTIALS;
+				goto send_reason_and_close;
 			case 5:
-				par->reason = LMQCP_REASON_UNSUPPORTED_PROTOCOL +
-						par->conn_rc - 1;
```

Comment:
I'm guessing this logic used to work at one point? Maybe the MQTT error code enum was updated to support 5.0 errors and that's where this broke.

Reply:
This approach should work for both 3.1.1 and 5.0